src/CqlBuilderPanel/functionsSection/FunctionsSection.tsx

```diff
@@ -3,26 +3,92 @@ import "./Functions.scss";
 import FunctionSectionNavTabs from "./FunctionSectionNavTabs";
 import Functions from "./functions/Functions";
 import FunctionBuilder from "./functionBuilder/FunctionBuilder";
+import {
+  CqlBuilderLookup,
+  Lookup,
+  FunctionLookup,
+} from "../..//model/CqlBuilderLookup";
+import * as _ from "lodash";
+import { CqlAntlr } from "@madie/cql-antlr-parser/dist/src";
 
-interface FunctionProps {
+export interface FunctionProps {
   canEdit: boolean;
-  handleApplyFunction: Function;
+  handleApplyFunction?: Function;
   loading: boolean;
+  cqlBuilderLookupsTypes?: CqlBuilderLookup;
+  cql: string;
+  isCQLUnchanged: boolean;
+  functions?: FunctionLookup[];
 }
+const getArgumentNames = (logic: string) => {
+  const args = logic.substring(logic.indexOf("(") + 1, logic.indexOf(")"));
+  return args.split(",");
+};
+const getFunctionLookups = (
+  lookups: Lookup[],
+  isFluent: string
+): FunctionLookup[] => {
+  return lookups.map((lookup) => {
+    return {
+      ...lookup,
+      isFluent: isFluent,
+      argumentNames: getArgumentNames(lookup.logic),
+    };
+  });
+};
 
 export default function FunctionsSection({
   canEdit,
   handleApplyFunction,
   loading,
+  cql,
+  isCQLUnchanged,
+  cqlBuilderLookupsTypes,
 }: FunctionProps) {
   const [activeTab, setActiveTab] = useState<string>("function");
 
+  const expressionDefinitions = cql
+    ? new CqlAntlr(cql).parse().expressionDefinitions
+    : [];
+
+  const functions: Lookup[] =
+    cqlBuilderLookupsTypes?.functions
+      ?.filter((func) => !func.libraryName)
+      .map((func) => {
+        // get the comments for CQL definition from antlr parser expressions
+        const expression = expressionDefinitions.find(
+          (expression) => func.logic == expression.text?.replace(/["']/g, "")
+        );
+        return { ...func, comment: expression?.comment };
+      }) || [];
+  let functionLookups: FunctionLookup[] = getFunctionLookups(functions, "-");
```

Comment:
these two can be combined.
 ```suggestion
  let functionLookups: FunctionLookup[] =
    cqlBuilderLookupsTypes?.functions
      ?.filter((func) => !func.libraryName)
      .map((func) => {
        // get the comments for CQL definition from antlr parser expressions
        const expression = expressionDefinitions.find(
          (expression) => func.logic == expression.text?.replace(/["']/g, "")
        );
        const args = getArgumentNames(func.logic);
        return {
          ...func,
          comment: expression?.comment,
          isFluent: "No",
          argumentNames: args,
        } as FunctionLookup;
      }) || [];
```

Reply:
@adongare Done.